### PR TITLE
Modified sample_pixel to read ds9 regions file

### PR DESCRIPTION
**Pull request by jmangum**
Since sample_position required knowing what the pixel position of the desired sample position was in the cutout cube, which one would not necessarily have before running CubeLineMoment, thought it useful to be able to read an (RA,Dec) position for sample_pixel from a ds9 file. CubeLineMoment now also extracts the label meta variable and uses it in the titles for the diagnostic plots.

Note, though, that I really do not like the way I solved this problem, as I had to read-in the cutoutcube in order to access its WCS information so that I could turn the (RA,Dec) from the ds9 file into a pixel coordinate. This extra FITS file read noticeably slows the script down. I am sure that @keflavich has a slicker way to get at the WCS information.

### CODE REVIEW

File: CubeLineMoment.py [hunk 1]
```diff
@@ -685,11 +693,31 @@ def main():
...
+    params['my_line_list'] = u.Quantity(parse_floatlist(params['my_line_list'].split(", ")), u.GHz)
     params['my_line_widths'] = u.Quantity(parse_floatlist(params['my_line_widths']), u.km/u.s)
     params['my_line_names'] = params['my_line_names'].split(", ")
     if 'sample_pixel' in params:
-        params['sample_pixel'] = ast.literal_eval(params['sample_pixel'])
+        params['sample_pixel'] != None
+#        params['sample_pixel'] = ast.literal_eval(params['sample_pixel'])
+        # Check to make sure that sample pixel regions file exists.  Open it if
+        #  it does exist, and exit script if it does not exist.
+        if os.path.isfile(params['sample_pixel']) == True:
```
Comment (keflavich) [inline CubeLineMoment.py]: `== True` is redundant

File: CubeLineMoment.py [hunk 2]
```diff
@@ -685,11 +693,31 @@ def main():
...
+            regsample = regions.read_ds9(params['sample_pixel'])
+        else:
+            print(params['sample_pixel'],' does not exist...terminating')
+            exit()
```
Comment (keflavich) [inline CubeLineMoment.py]: I don't think the use of `exit` is encouraged; raising an Exception might be more clear.

File: CubeLineMomentManual.md [hunk 3]
```diff
@@ -49,9 +49,10 @@ To run in ipython use:
 
 **spatial_mask_limit [float]:** Multiplier for noise-based spatial masking. Signal less than spatial_mask_limit times RMS noise is masked. Example: 2
 
+**sample_pixel [str, optional]:** A set of (x,y) coordinates to sample from the cutout cube to create diagnostic images.  Assumed to be in (RA,Dec) coordinates in a ds9 regions file, and must be within the cutout image area.  If left as `None`, no diagnostic images will be made.
```
Comment (keflavich) [inline CubeLineMomentManual.md]: if it's in a region file, it can be in any coordinate system

**Comment by keflavich** [comment]
@jmangum you should `git pull` the changes I just made, which resolved the conflict w/master

**Comment by jmangum** [comment]
Done.  Thanks.

File: CubeLineMoment.py (hunk 1 above)
Comment (jmangum) [reply CubeLineMoment.py]: Thanks.
`#        params['sample_pixel'] = ast.literal_eval(params['sample_pixel'])
        # Check to make sure that sample pixel regions file exists.  Open it if
        #  it does exist, and exit script if it does not exist.
        if os.path.isfile(params['sample_pixel']):`

File: CubeLineMomentManual.md (hunk 3 above)
Comment (jmangum) [reply CubeLineMomentManual.md]: Correct.  Can be any coordinate system.  Will update docs.

File: CubeLineMoment.py (hunk 2 above)
Comment (jmangum) [reply CubeLineMoment.py]: Agreed.  Changed to raised exception.

**Pull request merged**